classes/admin/class-submenu.php

```diff
@@ -701,36 +712,21 @@ protected function processActionCommand($id, $action, $command)
             : false;
 
         $builder = null;
-
         // transaction needs to be active (not in closed batch) to reverse.
         // gift service method for refund is reverse.
         if ($transaction->transactionStatus === 'A'
-            || $service instanceof HpsFluentGiftCardService
+            || $service instanceof GiftCard
         ) {
-            $builder = $service->reverse();
+            $builder = Transaction::fromId($transaction->transactionId)
```

Comment:
For gift transactions, we need to supply the `PaymentMethodType::GIFT` as the second parameter to `Transaction::fromId` in order for the request type to be mapped correctly for any non-credit card requests.

Reply:
This would apply to eCheck/ACH transactions as well.

Reply:
updated